web/src/lib/utils/retry.ts

```diff
@@ -0,0 +1,30 @@
+export const retry = <R, A = unknown>(
```

Comment:
I think you don't need to default `A = unknown`. TS should be able to infer that.